crates/ruff/src/rules/flake8_simplify/rules/ast_if.rs

```diff
@@ -455,6 +487,205 @@ fn compare_expr(expr1: &ComparableExpr, expr2: &ComparableExpr) -> bool {
     expr1.eq(expr2)
 }
 
+fn should_proceed_main(test: &Expr, body: &[Stmt], orelse: &[Stmt]) -> bool {
+    if let ExprKind::Compare {
+        left,
+        ops,
+        comparators,
+    } = &test.node
+    {
+        if let ExprKind::Name { .. } = &left.node {
+            if ops.len() == 1 && ops[0] == Cmpop::Eq {
+                if comparators.len() == 1 {
+                    if let ExprKind::Constant { .. } = &comparators[0].node {
+                        if body.len() == 1 {
+                            if let StmtKind::Return { .. } = &body[0].node {
+                                if orelse.len() == 1 {
+                                    if let StmtKind::If { .. } = &orelse[0].node {
+                                        return true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+fn should_proceed_child(stmt: &Stmt, var_id: &str) -> bool {
+    if let StmtKind::If { test, body, orelse } = &stmt.node {
+        if let ExprKind::Compare {
+            left,
+            ops,
+            comparators,
+        } = &test.node
+        {
+            if let ExprKind::Name { id, .. } = &left.node {
+                if id == var_id && ops.len() == 1 && ops[0] == Cmpop::Eq {
+                    if comparators.len() == 1 {
+                        if let ExprKind::Constant { .. } = &comparators[0].node {
+                            if body.len() == 1 {
+                                if let StmtKind::Return { .. } = &body[0].node {
+                                    if orelse.len() == 1 {
+                                        return true;
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    false
+}
+
+fn constant_to_str(value: &Constant) -> String {
+    match value {
+        Constant::None => "None".to_string(),
+        Constant::Str(s) => format!("\"{s}\""),
```

Comment:
what if `s` contains a `"`?

Reply:
Thank you for brining this up. @charliermarsh how do you want me to handle this? Usually we just detect what the type of quotation mark is at the beginning, and use that everywhere. But, this will not work here because if there is a tuple, we will have no way of knowing what the quotation marks were. I say the two options here are:
1. The "lazy" way: just format strings in debug (I implemented this already).
    - Pro: Already implemented, a fixer could switch the quotation marks back
    - Con: All inner double quotations would now be escaped in the users code
2. The "hard" way: write some code to figure out the starting parenthesis for each tuple item.
    - Pro: Users would have strings formatted EXACLTY as they want them
    - Con: Since rust_python does not give us locations for each item, we would need to find ourselves. Which probably involves going to tokens and then back

Reply:
does `unparse_expr(expr, checker.stylist)` help? tbh I don't understand what the code does.

Reply:
The issue would be that each item in the tuple could have a different type of quotation mark.

Reply:
We should try to use the `unparse_constant` method of `Generator` for this rather than implementing `constant_to_str` here. But separately, it's fine to pass `checker.stylist` in there and just use the quotation mark that was detected. If the user is mixing quotation marks anyway, it's fine if we modify some of them -- it's kind of like, if the user is mixing indentation styles, we won't respect that either.

(Is there any reason that changing the quotation marks would be a problem, apart from that it might surprise some users?)

Reply:
The issue I see is with a tuple like `("hello", 'I say "hi"')` -> `("hello", "I say "hi"")`

Reply:
Will try to review this soon, not tonight but maybe tomorrow. Sorry!

Reply:
Your merging an entire auto formatter, so ABSLOTELY no worries!! And, thank you so much for all the hard work. Can't wait until my CI is just:
```pip install ruff==1.0.0
ruff .
```